gitlab/v4/objects/registry_protection_rules.py

```diff
@@ -0,0 +1,28 @@
+from gitlab.base import RESTManager, RESTObject
+from gitlab.mixins import CreateMixin, ListMixin, SaveMixin, UpdateMethod, UpdateMixin
+from gitlab.types import RequiredOptional
+
+__all__ = [
+    "ProjectRegistryProtectionRule",
+    "ProjectRegistryProtectionRuleManager",
+]
+
+
+class ProjectRegistryProtectionRule(SaveMixin, RESTObject):
+    _repr_attr = "repository_path_pattern"
+
+
+class ProjectRegistryProtectionRuleManager(
+    ListMixin, CreateMixin, UpdateMixin, RESTManager
+):
+    _path = "/projects/{project_id}/registry/protection/rules"
+    _obj_cls = ProjectRegistryProtectionRule
+    _from_parent_attrs = {"project_id": "id"}
+    _create_attrs = RequiredOptional(
```

Comment:
What about `_update_attrs`?

Reply:
Yikes, I'll add these, as well as for packages, I might have missed that there as well.

Reply:
Added and did a little fixup I just noticed for packages in e0fa822b1beb45e3dc9a164b62824d3f9d544158.